src/components/Notifications/ModerationOfferModal.tsx

```diff
@@ -0,0 +1,128 @@
+import { alert } from "swal_config";
+import { patch } from "requests";
+
+import { errorAlerter } from "misc";
+
+interface Events {}
+
+interface ModerationOfferModalProperties {
+    player_id: number;
+    offered_powers: number; // Must be the full bitfield that the person is going to get
+    onResolved: () => void;
+}
+
+export class ModerationOfferModal extends Modal<Events, ModerationOfferModalProperties, {}> {
+    constructor(props) {
+        super(props);
+    }
+
+    accept = () => {
+        void alert.fire({
+            text: "Accepting...",
+            icon: "info",
+            showCancelButton: false,
+            showConfirmButton: false,
+            allowEscapeKey: false,
+        });
+
+        patch("me/moderation", {
+            moderator_powers: this.props.offered_powers,
+        })
+            .then(() => {
+                alert.close();
+                this.props.onResolved();
+            })
+            .catch(errorAlerter);
+        this.close();
+    };
+
+    reject = () => {
+        void alert.fire({
+            text: "OK, noted...",
+            icon: "info",
+            showCancelButton: false,
+            showConfirmButton: false,
+            allowEscapeKey: false,
+        });
+
+        patch("me/moderation", {
+            mod_powers_rejected: true,
+        })
+            .then(() => {
+                alert.close();
+                this.props.onResolved();
+            })
+            .catch(errorAlerter);
+        this.close();
+    };
+
+    render() {
+        return (
+            <div className="Modal ModerationOfferModal">
+                <div className="header">
+                    {pgettext(
+                        "Header telling people about community moderation powers they are offered",
+                        "Community Moderation",
+                    )}
+                </div>
+                <div className="moderation-offer-details">
+                    <p>
+                        {pgettext(
+                            "Part of the description offering community moderation powers to a user.",
+                            "If you're willing, we'd love to have you on board as a community moderator.",
+                        )}
+                    </p>
+                    <p>
+                        {pgettext(
+                            "Part of the description offering community moderation powers to a user.",
+                            "You'll get access to tools that will allow you to vote on reports raised by users - that will tell us how to handle those reports.",
+                        )}
+                    </p>
+                    <p>
+                        {pgettext(
+                            "Part of the description offering community moderation powers to a user.",
+                            "We just need your agreement to use the powers in the best interests of the OGS community, and in line with current policies and practices.",
```

Comment:
We should have a link to said policies and practices.. and which I guess means writing such things too :sweat_smile:

Reply:
That's why I wrote "practices" as well as "policies" 😝   The assumption being that anyone getting this kinda knows the deal even if it isn't written down formally 😅